mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.entity;
+
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.squareup.javapoet.TypeName;
+import java.beans.Introspector;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+public class DefaultEntityFactory implements EntityFactory {
+
+  private final ProcessorContext context;
+  // Note: the processor code is not concurrent, we only use ConcurrentMap for computeIfAbsent
```

Comment:
We cannot use java >= 1.8 to have `computeIfAbsent` on the `Map` interface?

Reply:
I don't understand. We do use java >= 1.8 (see parent POM), so `computeIfAbsent` is available.

Reply:
So we can use simple `HashMap`:
```
Map<String, String> map = new LinkedHashMap<>();
map.computeIfAbsent()
```
right?

Reply:
Ah right, for some reason I thought it was only on `ConcurrentMap` even in 1.8. Yes, we can use `HashMap`.

Reply:
Moot point now that the factory doesn't cache anymore, but thanks for the tip.